### PR TITLE
support for API 2200 for Storage and other resources

### PR DESCRIPTION
**Pull request by AsisBagga**
### Description
Support to below resources for API 2200
Storage Volume Template
Storage Volume
Storage System
Storage Pool
Server Profile
Server Profile Template

### Issues Resolved
[List any issues this PR will resolve. e.g., Fixes #01]

### Check List
- [ ] New functionality includes testing.
  - [x] All tests pass for go 1.11 + gofmt checks.
- [ ] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [x] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: examples/server_profile.go [hunk 1]
```diff
@@ -20,15 +20,15 @@ func main() {
...
 		"*")
 
 	initialScopeUris := new([]utils.Nstring)
-	scp, scperr := ovc.GetScopeByName("test")
+	scp, scperr := ovc.GetScopeByName("ScopeTest")
 	if scperr != nil {
 		*initialScopeUris = append(*initialScopeUris, scp.URI)
 	}
-	serverName, err := ovc.GetServerHardwareByName("0000A66102, bay 5")
+	serverName, err := ovc.GetServerHardwareByName("0000A66101, bay 5")
```
Comment (nabhajit-ray) [inline examples/server_profile.go:31]: Declare var for scope name, hardware etc

File: examples/storage_system.go [hunk 2]
```diff
@@ -20,7 +20,7 @@ func main() {
 		os.Getenv("ONEVIEW_OV_DOMAIN"),
 		os.Getenv("ONEVIEW_OV_ENDPOINT"),
 		false,
-		2000,
+		2200,
 		"*")
 
 	// Create storage system
```
Comment (nabhajit-ray) [inline examples/storage_system.go:26]: Remove username and password

File: examples/storage_volume.go [hunk 3]
```diff
@@ -21,26 +21,26 @@ func main() {
...
-		DataProtectionLevel: "NetworkRaid10Mirror2Way",
+		Name:             new_volume,
+		Storagepool:      st_pool.URI,
+		Size:             268435456,
+		ProvisioningType: "Thin",
+		//	DataProtectionLevel: "NetworkRaid10Mirror2Way",
 	}
 	trueVal := true
-	vol_template, err := ovc.GetStorageVolumeTemplateByName("template1")
+	vol_template, err := ovc.GetStorageVolumeTemplateByName("template")
```
Comment (nabhajit-ray) [inline examples/storage_volume.go:37]: Use Var instead of template name

File: examples/storage_volume_template.go [hunk 4]
```diff
@@ -184,10 +185,10 @@ func main() {
 		TemplateProperties: &Properties,
 		Name:               name_to_create,
 		Description:        "Volume template Example",
-		RootTemplateUri:    "/rest/storage-volume-templates/727d5e9a-9ca5-4788-8eb1-ac30011285e7",
+		RootTemplateUri:    "/rest/storage-volume-templates/6ae24a18-59a5-498b-9426-ac6200baa9e6",
```
Comment (nabhajit-ray) [inline examples/storage_volume_template.go:188]: Get root template

File: examples/server_profile_template.go [hunk 5]
```diff
@@ -31,7 +31,7 @@ func main() {
 		ManageConnections: true,
 	}
 	initialScopeUris := new([]utils.Nstring)
-	scp, scperr := ovc.GetScopeByName("test")
+	scp, scperr := ovc.GetScopeByName("ScopeTest")
```
Comment (nabhajit-ray) [inline examples/server_profile_template.go:34]: Use var instead of name

File: ov/storage_volume_template.go [hunk 6]
```diff
@@ -232,6 +232,7 @@ func (c *OVClient) UpdateStorageVolumeTemplate(sVolTemplate StorageVolumeTemplat
 
 	log.Debugf("REST : %s \n %+v\n", uri, sVolTemplate)
 	_, err := c.RestAPICall(rest.PUT, uri, sVolTemplate)
+
```
Comment (nabhajit-ray) [inline ov/storage_volume_template.go:235]: Unnecessary changes. Why empty line is added?

File: examples/server_profile.go (hunk 1 above)
Comment (AsisBagga) [reply examples/server_profile.go]: done

File: examples/storage_system.go (hunk 2 above)
Comment (AsisBagga) [reply examples/storage_system.go:26]: you mean the for below line?
storageSystem := ov.StorageSystem{Hostname: "172.18.30.1", Username: "dcs", Password: "dcs", Family: "StoreVirtual", Description: "<description>"}

File: examples/storage_volume.go (hunk 3 above)
Comment (AsisBagga) [reply examples/storage_volume.go]: done

File: examples/storage_volume_template.go (hunk 4 above)
Comment (AsisBagga) [reply examples/storage_volume_template.go:188]: can you share the endpoint for getting root template?

File: examples/server_profile_template.go (hunk 5 above)
Comment (AsisBagga) [reply examples/server_profile_template.go]: done

File: examples/storage_volume.go [hunk 7]
```diff
@@ -3,7 +3,7 @@ package main
 import (
 	"fmt"
 	"github.com/HewlettPackard/oneview-golang/ov"
-	"github.com/HewlettPackard/oneview-golang/utils"
+	//	"github.com/HewlettPackard/oneview-golang/utils"
```
[the remark this answers is not in the document]
Comment (AsisBagga) [reply examples/storage_volume.go]: removed.

File: examples/storage_volume.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (AsisBagga) [reply examples/storage_volume.go]: done

File: ov/storage_volume_template.go (hunk 6 above)
Comment (AsisBagga) [reply ov/storage_volume_template.go]: removed.

File: examples/storage_system.go (hunk 2 above)
Comment (nabhajit-ray) [reply examples/storage_system.go:26]: yes, also use variable here also

File: examples/storage_volume_template.go (hunk 4 above)
Comment (nabhajit-ray) [reply examples/storage_volume_template.go:188]: Add functionality in library files. Filter by field where isRoot is true

File: examples/storage_system.go (hunk 2 above)
Comment (asisbagga-dev) [reply examples/storage_system.go:31]: done

File: examples/storage_volume_template.go (hunk 4 above)
Comment (asisbagga-dev) [reply examples/storage_volume_template.go]: done

File: examples/storage_system.go (hunk 2 above)
Comment (AsisBagga) [reply examples/storage_system.go:31]: okay, have updated.

File: examples/storage_volume_template.go (hunk 4 above)
Comment (AsisBagga) [reply examples/storage_volume_template.go]: done

**Review by nabhajit-ray — APPROVED** [review]

**Review by chebroluharika — APPROVED** [review]
